src/hooks/useConditionalEffect/useConditionalEffect.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { act } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+
+import { renderHookSSR } from '../../_internal/test-utils/renderHookSSR.tsx';
+
+import { useConditionalEffect } from './useConditionalEffect.ts';
+
+describe('useConditionalEffect', () => {
+  it('is safe on server side rendering', () => {
+    const effect = vi.fn();
+    const condition = vi.fn(() => false);
+
+    renderHookSSR.serverOnly(() => useConditionalEffect(effect, [1], condition));
+
+    expect(condition).toHaveBeenCalled();
+    expect(effect).not.toHaveBeenCalled();
+  });
+
+  it('should not run effect when condition returns false', async () => {
+    const effect = vi.fn();
+    const condition = vi.fn(() => false);
+
+    await renderHookSSR(() => useConditionalEffect(effect, [1], condition));
+
+    expect(condition).toHaveBeenCalledWith(undefined, [1]);
+    expect(effect).not.toHaveBeenCalled();
+  });
+
+  it('should run effect when condition returns true', async () => {
+    const effect = vi.fn();
+    const condition = vi.fn(() => true);
+
+    await renderHookSSR(() => useConditionalEffect(effect, [1], condition));
+
+    expect(condition).toHaveBeenCalledWith(undefined, [1]);
+    expect(effect).toHaveBeenCalled();
+  });
+
+  it('should warn when an empty dependency array is provided', async () => {
+    const originalWarn = console.warn;
+    console.warn = vi.fn();
+
+    const effect = vi.fn();
+    const condition = vi.fn(() => true);
+
+    await renderHookSSR(() => useConditionalEffect(effect, [], condition));
+
+    expect(console.warn).toHaveBeenCalledWith(
+      'useConditionalEffect received an empty dependency array. ' +
+        'This may indicate missing dependencies and could lead to unexpected behavior.'
+    );
+
+    console.warn = originalWarn;
+  });
+
+  it('should run effect multiple times when condition is repeatedly true', async () => {
+    const effect = vi.fn();
+    const condition = vi.fn(() => true);
+
+    const { rerender } = await renderHookSSR(({ deps }) => useConditionalEffect(effect, deps, condition), {
+      initialProps: { deps: [1] as const },
```

Comment:
This code results in a type error.
Is the const assertion strictly necessary?

```suggestion
      initialProps: { deps: [1] },
```